### PR TITLE
Make toc more usable

### PR DESCRIPTION
**Pull request by ru-fu**
See the different commits for an explanation of the changes.

### CODE REVIEW

**Comment by toby63** [comment]
I did this before, here is the reason why I used a manually created TOC instead:
When you limit the TOC, some categories don't get indexed, so you can't link to them anymore.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -24,7 +24,10 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -y man2html-base pycodestyle python3-bs4 python3-markdown
+          sudo apt-get install -y man2html-base pycodestyle python3-bs4
+
+      - name: Install Python dependencies
+        run: pip3 install -r requirements.txt
```
Comment (stgraber) [inline .github/workflows/test.yml:30]: @ru-fu I think you may need to run that under `sudo` so it's globally available for the next stage.

Comment (ru-fu) [reply .github/workflows/test.yml]: Thanks! That saved me a lot of debugging. :+1:

**Comment by ru-fu** [comment]
I want to move the ToC to the side as well, but let's leave that for a separate PR.

**Comment by toby63** [comment]
Maybe you could add a change for the Contributing document as well:

https://github.com/lxc/linuxcontainers.org/blob/master/CONTRIBUTING.md#guidelines-for-the-content

I added this text in the past, that would now need to be changed:
`create a Table of Content (TOC) manually (with a markdown list), the toc extension is only used for creation of hyperlinks.`

Maybe to the following:
`create a Table of Content (TOC) (if applicable) with the [TOC] tag on top of the page`

**Comment by stgraber** [comment]
@ru-fu can you update CONTRIBUTING.md? After that, I think we'll be good to merge but I'll have to take a look at the server that builds the website to make sure it's got a suitable version of markdown.

**Comment by ru-fu** [comment]
> Maybe you could add a change for the Contributing document as well:

Thanks, good catch! Updated now.

> but I'll have to take a look at the server that builds the website to make sure it's got a suitable version of markdown.

Oh right - I thought we were publishing directly through GitHub.

**Pull request merged**